pkg/cli-runtime/testing/validatable_table.go

```diff
@@ -60,24 +72,31 @@ func (ts ValidatableTestSuite) Run(t *testing.T) {
 	}
 	if len(focused) != 0 {
 		for _, tc := range focused {
-			tc.Run(t)
+			tc.Run(t, ctx)
 		}
 		t.Errorf("test run focused on %d record(s), skipped %d record(s)", len(focused), len(ts)-len(focused))
 		return
 	}
 
 	for _, tc := range ts {
-		tc.Run(t)
+		tc.Run(t, ctx)
 	}
 }
 
-func (tc ValidatableTestCase) Run(t *testing.T) {
+func (tc ValidatableTestCase) Run(t *testing.T, ctx context.Context) {
 	t.Run(tc.Name, func(t *testing.T) {
 		if tc.Skip {
 			t.SkipNow()
 		}
 
-		errs := tc.Validatable.Validate(context.Background())
```

Comment:
+1

Reply:
so when I was trying to create the validatable cases for this, every time the case got to the this part 

```
if opts.UpdateStrategy != "" && cli.CommandFromContext(ctx).Flags().Changed(cli.StripDash(flags.UpdateStrategyFlagName))
```

Since there was just an empty context, there were no flags and I was getting a `nil` error . I needed to add the context in order to set the command in the test during the `preprare` stage

Reply:
if you can pass a fresh context to prepare and use that for validate as well as cleanup for each test case the error shouldn't happen. If we use the same context across all the test cases under a suite, wouldnt it hinder test cases that doesnt have a cleanup ?